File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
    const map = new Map();
    
    for(let p of participant){
        map.set(p, (map.get(p) || 0) + 1);
    }
    
    for(let c of completion){
        map.set(c, map.get(c) - 1);
    }
    
    for(const [key, value] of map){
        if(value > 0){
            return key;
        }
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        HashMap<String, Integer> hm = new HashMap<>();
        
        for(int i = 0; i < participant.length; i++){
            hm.put(participant[i], hm.getOrDefault(participant[i], 0) + 1);
        }
        
        for(int i = 0; i < completion.length; i++){
            hm.put(completion[i], hm.get(completion[i]) - 1);
        }
        
        for(String name : hm.keySet()){
            if(hm.get(name) != 0){
                return name;
            }
        }
        
        return participant[participant.length - 1];
    }
}